dask_ml/model_selection/_successive_halving.py

```diff
@@ -0,0 +1,134 @@
+import copy
+import math
+from collections import defaultdict
+
+import numpy as np
+import toolz
+
+from ._incremental import INC_ATTRS, IncrementalSearchCV
+
+DOC = (
+    """
+    Perform the successive halving algorithm.
+
+    Parameters
+    ----------
+    estimator : estimator object.
+        A object of that type is instantiated for each initial hyperparameter
+        combination. This is assumed to implement the scikit-learn estimator
+        interface. Either estimator needs to provide a `score`` function,
+        or ``scoring`` must be passed. The estimator must implement
+        ``partial_fit``, ``set_params``, and work well with ``clone``.
+
+    param_distributions : dict
+        Dictionary with parameters names (string) as keys and distributions
+        or lists of parameters to try. Distributions must provide a ``rvs``
+        method for sampling (such as those from scipy.stats.distributions).
+        If a list is given, it is sampled uniformly.
+
+    n_initial_parameters : int, default=10
+        Number of parameter settings that are sampled.
+        This trades off runtime vs quality of the solution.
+
+        Alternatively, you can set this to ``"grid"`` to do a full grid search.
+
+    resource : int
+        Number of times to call partial fit initially. Estimators are trained
+        for ``resource`` calls to ``partial_fit`` at first, then additional
+        times increasing by ``aggressivenes``.
+
+    aggressizeness : float, default=3
+        How aggressive to be in culling off the models. Higher
+        values correspond to being more aggressive in killing off
+        models. The "infinite horizon" theory suggests ``aggressizeness == np.e``
+        is optimal.
+
+    kwargs : dict
+        Parameters to pass to
+        :class:`~dask_ml.model_selection.IncrementalSearchCV`.
+
+    """
+    + INC_ATTRS
+)
+
+
+class SuccessiveHalvingSearchCV(IncrementalSearchCV):
+    __doc__ = DOC
+
+    def __init__(
+        self,
+        estimator,
+        param_distribution,
+        n_initial_parameters=15,
+        start_iter=9,
+        aggressiveness=3,
+        limit=None,
```

Comment:
It'd be nice to remove `limit` or have a better name.

Reply:
`max_iter` is the right word for that.

Reply:
But `max_iter` is already used to indicate how the maximum iterations for the estimator. I'm using `adaptive_max_iter` instead.